config.js

```diff
@@ -238,114 +238,130 @@ exports.updateFromArgv = function() {
-    }
-
-    if (argv.nopause && (argv.nopauseranked || argv.nopauseunranked)) {
-        console.log("Warning: You are using --nopause in combination with --nopauseranked and/or --nopauseunranked. \n Use either --nopause alone, OR --nopauseranked with --nopauseunranked.\nBut don't use the 3 nopause arguments at the same time.");
-    }
-
-    if (argv.komi) {
-        if (argv.komi.includes(`auto`)) {
-            console.log("Warning: /--komi auto/ has been renamed to /--komi automatic/\n");
-            // we skip a line here, not below, because this argv may be undefined if not used by bot admin
+    function depreciatedArgumentIsUsedNewArgumentToUseInstead(depreciatedArgumentUsed, newArgumentToUseInstead, indicatorDepreciation) {
+        if (depreciatedArgumentUsed) {
+            indicatorDepreciation = true;
+            console.log(`Warning: --${depreciatedArgumentUsed} is no longer supported. Use --${newArgumentToUseInstead} instead.\nRead latest up to date available gtp2ogs arguments in OPTIONS-LIST (devel branch) for details : https://github.com/online-go/gtp2ogs/blob/devel/docs/OPTIONS-LIST.md`);
         }
     }
 
-    if (argv.komiranked) {
-        if (argv.komiranked.includes(`auto`)) {
-            console.log("Warning: /--komiranked auto/ has been renamed to /--komiranked automatic/\n");
-            // we skip a line here, not below, because this argv may be undefined if not used by bot admin
+    depreciatedArgumentIsUsedNewArgumentToUseInstead(argv.botid, argv.username, this.indicatorConsoleDepreciation);
+    depreciatedArgumentIsUsedNewArgumentToUseInstead(argv.bot, argv.username, this.indicatorConsoleDepreciation);
+    depreciatedArgumentIsUsedNewArgumentToUseInstead(argv.id, argv.username, this.indicatorConsoleDepreciation);
+    depreciatedArgumentIsUsedNewArgumentToUseInstead(argv.minrankedhandicap, argv.minhandicapranked, this.indicatorConsoleDepreciation);
+    depreciatedArgumentIsUsedNewArgumentToUseInstead(argv.minunrankedhandicap, argv.minhandicapunranked, this.indicatorConsoleDepreciation);
+    depreciatedArgumentIsUsedNewArgumentToUseInstead(argv.maxrankedhandicap, argv.maxhandicapranked, this.indicatorConsoleDepreciation);
+    depreciatedArgumentIsUsedNewArgumentToUseInstead(argv.maxunrankedhandicap, argv.maxhandicapunranked, this.indicatorConsoleDepreciation);
+    depreciatedArgumentIsUsedNewArgumentToUseInstead(argv.maxtotalgames, argv.maxconnectedgames, this.indicatorConsoleDepreciation);
+    depreciatedArgumentIsUsedNewArgumentToUseInstead(argv.maxactivegames, argv.maxconnectedgamesperuser, this.indicatorConsoleDepreciation);
+
+    function depreciatedArgumentIsUsedNewArgumentToUseInsteadUpToThree(depreciatedArgumentUsed, newArgumentToUseInsteadOne, newArgumentToUseInsteadTwo, newArgumentToUseInsteadThree, indicatorDepreciation) {
```

Comment:
I think in this case you want something like:

```
function deprecatedArgumentAlternatives(deprecatedArg, alternatives) {
  console.log("Warning: --${deprecatedArg} is no longer supported. Use one of",
      alternatives.map(x => ("--" + x)).join(" ,"),
      "instead.";
  )
}
```

To avoid having separate helper functions for arguments with different numbers of alternatives.

Reply:
a bit complicated for me to understand it, as i dont know all the theory behind it :+1:

Reply:
You can also create the alternatives part of the string explicitly, which might be better:

```
let msg = "Warning: --${deprecatedArg} is no longer supported. Use one of ";
let i = 0;
for (let e of alternatives) {
  if (i == 0) {
    msg += "--" + e;
  } else {
    msg += ", --" + e;
  }
msg += ".";
console.log(msg);
```

The `map()` followed by `join()` is just a shorter way of expressing the same:

`map(x => ("--" + x)` takes each element of an `Array` and (assuming it is a string) prepends it with a "--".

`join(", ")` takes all elements of an `Array` and joins them together into one string, with ", " separating each element.

Doing certain simple transformations on lists (`filter`, `map`, `reduce`) is a pretty common approach, see for example this blog post to illustrate:

https://medium.freecodecamp.org/ruby-using-the-select-map-and-reduce-methods-together-a9b2af30804b

But as you correctly point out, it can be harder to read the more complicated it gets. And maybe in this case it was already too much.

Reply:
i see, its much clearer now :+1: @windo 

but more importantly,
how can i extract the same from a variable

for example extract `--maxtotalgames` from `argv.maxtotalgames=20` is no longer supported

Reply:
I think you have to pass in `"maxtotalgames"` as a string, rather than the variable, that is:

```
checkDeprecation("maxtotalgames");
```

And not:

```
checkDeprecation(argv.maxtotalgames);
```